hil-test/tests/lcd_cam_i8080.rs

```diff
@@ -47,14 +45,14 @@ mod tests {
         let pcnt = Pcnt::new(peripherals.PCNT);
         let io = Io::new(peripherals.GPIO, peripherals.IO_MUX);
         let (_, _, tx_buffer, tx_descriptors) = dma_buffers!(0, DATA_SIZE);
+        let dma_buf = DmaTxBuf::new(tx_descriptors, tx_buffer).unwrap();
```

Comment:
I'm starting to think that we could turn `new` into an infallible function (that panics), and add `try_new` that returns a `Result`. Relying so heavily on `unwrap` is a problem because if the compiler can't prove that the constructor doesn't fail, it will include the core panic formatting unless the user rolls their own `unwrap!` for defmt.

Reply:
I'm definitely not against this. I want to take this even further and just have this.

```Rust
let dma_buf = dma_tx_buf!(2000);
```

> if the compiler can't prove that the constructor doesn't fail, it will include the core panic formatting

TIL, I'm guessing this is undesirable. I'm not too familiar with this area in the embedded Rust space.
It's worth getting it into the API-GUIDELINES for people like me who don't know about it

Reply:
We could probably start doing it before tbh, once the dust settles on the current dma macros.

Reply:
I've been thinking about this too. I guess to add to "optimise for the common case", what can you actually do other than `unwrap()` if the requirements of the slice isn't met? You have to change the program (unless using an allocator, but if you got unaligned memory before, chances are you aren't asking the allocator for the right thing either). If 99% of the time you are unwrapping, it's probably a good candidate for an internal panic.